src/test/HostActivationTests/NativeHosting/Nethost.cs

```diff
@@ -142,6 +137,57 @@ public void GetHostFxrPath_HostFxrAlreadyLoaded()
                 .And.HaveStdErrContaining($"Found previously loaded library {HostFxrName}");
         }
 
+        [Theory]
+        [InlineData("{0}", true)]
+        [InlineData("{0}\n", true)]
+        [InlineData("{0}\nSome other text", true)]
+        [InlineData("", false)]
+        [InlineData("\n{0}", false)]
+        [InlineData(" {0}", false)]
+        [InlineData("{0} \n", false)]
+        [InlineData("{0} ", false)]
+        public void GetHostFxrPath_InstallLocationFile(string value, bool shouldPass)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // This test targets the install_location config file which is only used on Linux and macOS.
+                return;
+            }
+
+            string installLocation = Path.Combine(sharedState.ValidInstallRoot, "dotnet");
+
+            using (RegisteredInstallLocationOverride registeredInstallLocationOverride = new RegisteredInstallLocationOverride())
+            {
+                File.WriteAllText(registeredInstallLocationOverride.PathValueOverride, string.Format(value, installLocation));
```

Comment:
`registeredInstallLocationOverride.SetInstallLocation`?

Reply:
That would probably work, but I intentionally used the explicit file write here because this is testing values which are not just locations. So basically this should be the only test which does not use the `SetInstallLocation`.